D/D.sublime-syntax

```diff
@@ -8,339 +8,2032 @@ file_extensions:
+          scope: punctuation.separator.mapping.d
+          set: [enum-body, type]
+        - match: '(?={)'
+          set: anonymous-enum-body
+        - match: '(?=\S)'
+          push: type
+  enum-after:
+    - include: end-of-line
+    - match: '(?={)'
+      set: enum-body
+    - match: ':'
+      scope: punctuation.separator.mapping.d
+      set: [enum-body, type]
+    - match: '(?==|\()'
+      set: enum-assignment
+    - match: '\b({{name}})\b'
+      scope: entity.name.enum.d
+    - include: expect-end-of-line
+
+  enum-body:
+    - match: '{'
+      scope: punctuation.section.block.begin.d
+      set:
+        - match: '}'
+          scope: punctuation.section.block.end.d
           pop: true
-    - match: (//).*$\n?
-      scope: comment.line.double-slash.d
+        - match: '(?=\b({{name}})\b)'
+          push: enum-member
+    - include: not-whitespace-illegal
+
+  enum-assignment:
+    - match: '='
+      scope: keyword.operator.assignment.d
+      set: [enum-assignment-after, value]
+    - match: '(?=\()'
+      push: function-argument-definition-list
+    - include: not-whitespace-illegal-pop
+  enum-assignment-after:
+    - include: end-of-line
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      set:
+        - match: '\b({{name}})\b'
+          scope: entity.name.enum.d
+          set: enum-assignment
+        - include: not-whitespace-illegal-pop
+    - include: not-whitespace-illegal-pop
+
+  anonymous-enum-body:
+    - match: '{'
+      scope: punctuation.section.block.begin.d
+      set:
+        - match: '}'
+          scope: punctuation.section.block.end.d
+          pop: true
+        - include: attribute-in
+        - match: '(?=\b({{name}})\s*(=|,|}|$))'
+          push: enum-member
+        - match: '{{type_look_ahead}}'
+          push: [enum-member, type]
+    - include: not-whitespace-illegal
+
+  enum-member:
+    - match: '\b({{name}})\b'
+      scope: entity.name.constant.d
+      set:
+        - match: '='
+          scope: keyword.operator.assignment.d
+          set: [enum-member-after, value]
+        - match: ','
+          scope: punctuation.separator.sequence.d
+          pop: true
+        - match: '(?=})'
+          pop: true
+        - include: not-whitespace-illegal
+    - include: not-whitespace-illegal
+  enum-member-after:
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      pop: true
+    - match: '(?=})'
+      pop: true
+    - include: not-whitespace-illegal-pop
+
+  conditional-declaration-in:
+    - match: '\b(version)\b'
+      scope: keyword.control.conditional.d
+      push: [optional-block-statement-or-label, version-condition]
+    - match: '\b(debug)\b'
+      scope: keyword.control.conditional.d
+      push: [optional-block-statement-or-label, debug-condition]
+
+  version-condition:
+    - match: '\('
+      scope: punctuation.section.parens.begin.d
+      set:
+        - match: '\b(unittest|assert)\b'
+          scope: keyword.d
```

Comment:
We should probably add a second label to this scope.

Reply:
In this case the keywords have no direct meaning. They're just scoped as keyword because that's what the lexer does, but otherwise they're no different here to a normal name.

Reply:
If they don't do anything, why highlight them? Or is this based on a naming convention? In that case, something like `support.function` might seem appropriate?

Reply:
Because that's what the D grammar defines them as, they *are* keywords. They're a special case of version statements because they check for when unittests or asserts are enabled, but they're no different from any regular version check in every other sense. It's just like `.class` in ruby, where it's a keyword specialcased to serve another function. I don't think that makes it less of a keyword.

Reply:
I think most users would expect that these keywords would be consistently colored, so I would use the same scope as they get in other parts of the syntax.